tobac/utils/general.py

```diff
@@ -631,6 +631,228 @@ def combine_feature_dataframes(
+
+    return features
+
+
+@internal_utils.iris_to_xarray
+def get_statistics_from_mask(
+    segmentation_mask: xr.DataArray,
+    *fields: xr.DataArray,
+    features: pd.DataFrame,
+    func_dict: dict[str, tuple[Callable]] = {"Mean": np.mean},
+    index: None | list[int] = None,
+    default: None | float = None,
+    id_column: str = "feature",
+) -> pd.DataFrame:
+    """
+    Derives bulk statistics for each object in the segmentation mask.
+
+
+    Parameters:
+    -----------
+    segmentation_mask : xr.DataArray
+        Segmentation mask output
+    *fields : xr.DataArray[np.ndarray]
+        Field(s) with input data. Needs to have the same dimensions as the segmentation mask.
+    features: pd.DataFrame
+        Dataframe with segmented features (output from feature detection or segmentation).
+        Timesteps must not be exactly the same as in segmentation mask but all labels in the mask need to be present in the feature dataframe.
+    func_dict: dict[str, Callable], optional (default: {'ncells':np.count_nonzero})
+        Dictionary with function(s) to apply over each region as values and the name of the respective statistics as keys
+        default is to just count the number of cells associated with each feature and write it to the feature dataframe
+    index: None | list[int], optional (default: None)
+        list of indexes of regions in labels to apply function to. If None, will
+            default to all integers between 1 and the maximum value in labels
+    default: None | float, optional (default: None)
+        default value to return in a region that has no values
+    id_column: str, optional (default: "feature")
+       Name of the column in feature dataframe that contains IDs that match with the labels in mask. The default is the column "feature".
+
+
+     Returns:
+     -------
+     features: pd.DataFrame
+         Updated feature dataframe with bulk statistics for each feature saved in a new column
+    """
+    # check that mask and input data have the same dimensions
+    for field in fields:
+        if segmentation_mask.shape != field.shape:
+            raise ValueError("Input labels and field do not have the same shape")
+
+    # warning when feature labels are not unique in dataframe
+    if not features.feature.is_unique:
+        raise logging.warning(
+            "Feature labels are not unique which may cause unexpected results for the computation of bulk statistics."
+        )
+
+    # get bulk statistics for each timestep
+    for tt in pd.to_datetime(segmentation_mask.time):
+        # select specific timestep
+        segmentation_mask_t = segmentation_mask.sel(time=tt).data
+        field_t = field.sel(time=tt).data
+
+        # make sure that the labels in the segmentation mask exist in feature dataframe
+        if (
+            np.intersect1d(np.unique(segmentation_mask_t), features.feature).size
+            > np.unique(segmentation_mask_t).size
+        ):
```

Comment:
I think this should be less than, rather than greater than
```suggestion
        if (
            np.intersect1d(segmentation_mask_t, features[id_column]).size
            < np.unique(segmentation_mask_t).size
        ):
```

However, segmentation does not necessarily create a label for every feature (e.g. if the seed location is below the segmentation threshold value), so we should be careful about raising an exception here as it may break segmentation